### PR TITLE
Add AWS tutorial that shows Elastic Agent

### PR DESCRIPTION
**Pull request by dedemorton**
Adds a new version of the AWS tutorial that works with Elastic Agent instead of Beats. We will continue to provide the Beats tutorial, but without updates.

Access the preview here: https://observability-docs_2106.docs-preview.app.elstc.co/guide/en/observability/master/monitor-aws-elastic-agent.html

What's left to do:
- [x]  Add a screen capture that shows the Logs app. I didn't do that yet because I see some "failed to find message" messages. Should I ignore and include a screenshot anyhow? _Decision:_ I'll filter on VPC Flow Logs to avoid showing the messages that include "failed to find message" .
image
- [x] ~Add a screen capture of the Infrastructure app. I need someone to help me with this because I am seeing a "There is no data to display" message. Under data streams, I do see a data stream called aws.ec2_metrics that is growing, but I don't see EC2 metrics in Discover or the Infrastructure app. Maybe I just don't have the right options selected? Or I haven't set things up correctly in AWS?~
_Decision:_ I'll show the Discover app instead filtered set to `data_stream.dataset:"aws.ec2_metrics"`. We can update the tutorial later to show the infra app.
After merging:
- [ ] If we want to backport to 8.3, I'll need to: change Infrastructure to Metrics. I'll also need to confirm that the screen captures are correct.

### CODE REVIEW

**Comment by dedemorton** [comment]
@ravikesarwani FYI...could use your input here, too. Shouldn't take long to finish this up...just ran out of time last week and wanted to get something up before my PTO. I just need to run through the steps on 8.4.1, update a few images, and do a light edit...then this should be ready. Unless I run into _more_bugs....

For background:
Ran into a [bug on 8.4.0](https://github.com/elastic/beats/issues/32888), so I had to use 8.3.3. Turns out that the AWS integration has a [couple dashboard bugs](https://github.com/elastic/integrations/issues/4082) (missing fields) that I had to work around. :-/  I want to test these docs again on 8.4.1 because the user will run into problems if they try to run this tutorial on 8.3.3 or 8.4.0.

**Comment by tommyers-elastic** [comment]
@dedemorton billing dashboard fixed in aws integration v1.23.3

#4082

**Comment by ravikesarwani** [comment]
> Edit the queue you created and use a JSON object to define an advanced access policy:

Maybe good to explain what the policy is doing. Something like "Edit the queue you created and use a JSON object to define an advanced access policy. The access policy allows S3 ObjectCreated events to be sent to the queue."

**Comment by ravikesarwani** [comment]
In the diagrams, for more clarity I would suggest naming the green S3 buckets to =>
S3 Bucket (VPC flow logs) & S3 Bucket (S3 access logs)
“VPC flow logs” and “S3 access logs” text can be in the second line with smaller fonts.

Same suggestion would be for the SQS blocks, if easily possible as well.

**Comment by ravikesarwani** [comment]
> Reviewers: My log stream has "failed to find message" under messages, so I am not including a screen capture yet. Is that expected? Have I done something wrong?

I am not sure but check the "settings" to make sure you have the logs-* selected in the Log indices.
Also in the Stack management->Index Management->Data streams you have logs-aws.vpcflow-default data stream where the actual VPC flows logs are being sent by the Elastic Agent.

**Comment by ravikesarwani** [comment]
DeDe, thank you so much for your great effort on this. I felt the flow was good and descriptive. I know it feels a little long but I will take the extra details any time of the day rather than having users not understand things clearly. Really appreciate it!
 
I wonder if there is any control to make some of the extra details "collapsed" by default so that in first view the tutorial doesn't look too long (and hence perceived complex to some users). They can expand the sections if needed (for example setup of SQS queue and objectcreated event notification are really AWS set up we are including to provide completion and maybe collapsed initially).

**Comment by ravikesarwani** [comment]
cc: @pmeresanu85 FYI ... just for awareness

**Comment by dedemorton** [comment]
This is ready for a proper review now.

**Review by ollyhowell — APPROVED** [review]
Thank you @dedemorton echoing Ravi's comments but really appreciate reading such thoughtful and clearly written instructions - great job! 

Looks good to me so approving in advance, with some minor comments and suggestions inline.

I had some general points/questions too:

- Screenshots: do you normally add the "Screenshot showing..." for accessibility? Didn't check all tags but seemed some did and some didn't, also no "Diagram showing..." equivalent
- Screenshots: is it also worth adding the aria description Colleen did recently?
- Regions: I saw the tip re: choosing same region as deployment - is this needed in a few other places (where regions were specified) or could we make expand on that in the first tip in terms of where it is needed (and possible side effects of mismatch)?
- I did have some minor comments for the Beats one but since the PR doesn't have those changes, what's the best way to add these? Was just one or two minor things (is Filebeat an agent? regions advice again, reference to "see" that could be "confirm", some interesting plurals ("data are", "configs"), and maybe a Congrats and next steps at end (similar to Agent version)
- Will leave discussion of Tutorials section (and these tutorials specifically) to side for now as it's probably cleaner and easier to consider separately / as part of ingest

**Review by ravikesarwani — APPROVED** [review]
Looks good from my side.

File: docs/en/observability/monitor-aws-agent.asciidoc [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+==== Create an SQS queue
+
+To create the SQS queue:
+
+. Go to the https://console.aws.amazon.com/sqs/[SQS console] and create an SQS
+queue. Create a standard SQS queue that uses the default settings.
++
+[IMPORTANT]
+=====
+Make sure you create the SQS queue in the same region as the S3 bucket.
```
[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-aws-agent.asciidoc:85]: I doesn't work at all if you create the queue in the wrong region.

File: docs/en/observability/monitor-aws-agent.asciidoc [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+**{aws}** source and the **EC2 Instances**.
+
+The AWS integration also comes with pre-built dashboards that you can use to
+visualize the data. In {kib}, open the main menu and click **Dashboard**. Search
+for EC2 and select the dashboard called **[Metrics AWS] EC2 Overview**:
+
+[role="screenshot"]
+image::images/agent-tut-ec2-overview-dashboard.png[Screenshot of Billing Overview dashboard]
+
+To track your billings on {aws}, open the
```
[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-aws-agent.asciidoc:429]: Yah good point. This was left over from the Beats version. It did not look weird to my American eyes, but I think "AWS billing" might sound better, so I will update accordingly.

Comment (dedemorton) [inline docs/en/observability/monitor-aws-agent.asciidoc:429]:
```suggestion
To track your AWS billing, open the
```

**Comment by dedemorton** [comment]
@olly Responses to your questions:

>Screenshots: do you normally add the "Screenshot showing..." for accessibility? Didn't check all tags but seemed some did and some didn't, also no "Diagram showing..." equivalent

Good catch. I think it's good to set the context as much as possible by describing the image. I will make the descriptions consistent.

>Screenshots: is it also worth adding the aria description Colleen did recently?

Not in this case. For accessibility, I’m always careful to describe the diagram content in the text that surrounds the diagram, so I don’t feel detailed aria descriptions are necessary here.

>Regions: I saw the tip re: choosing same region as deployment - is this needed in a few other places (where regions were specified) or could we make expand on that in the first tip in terms of where it is needed (and possible side effects of mismatch)?

Not sure, but I'll take a look. Once users are logged into a specific region, I don't think there will be a problem with them creating objects in the wrong region. I also think experienced AWS users are probably more aware of the limitations across regions than we folks who write tutorials. ;-P

>I did have some minor comments for the Beats one but since the PR doesn't have those changes, what's the best way to add these? Was just one or two minor things (is Filebeat an agent? regions advice again, reference to "see" that could be "confirm", some interesting plurals ("data are", "configs"), and maybe a Congrats and next steps at end (similar to Agent version)
Will leave discussion of Tutorials section (and these tutorials specifically) to side for now as it's probably cleaner and easier to consider separately / as part of ingest

I think we should limit our effort with the existing tutorials, but you can always open a PR if you have some quick fixes. I want to keep this PR focused on the Elastic Agent version of the tutorial. Otherwise we risk opening a can of worms.

File: docs/en/observability/monitor-aws-agent.asciidoc [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+[discrete]
+[[aws-elastic-agent-collect-metrics]]
+=== Step 6: Collect {aws} metrics
+
+In this step, you configure the AWS integration to periodically fetch monitoring
+metrics from AWS CloudWatch using **GetMetricData** API for {aws} services.
+Specifically you'll learn how to stream and process billing and EC2 metrics.
+
+IMPORTANT: Extra AWS charges on CloudWatch API requests may be generated if you
```
[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-aws-agent.asciidoc:367]: Good idea! I can add a link to this page: https://aws.amazon.com/cloudwatch/pricing/

[the remark this answers is not in the document]
Comment (dedemorton) [reply docs/en/observability/monitor-aws-agent.asciidoc:373]: @ollyhowell AFAIK, the integrations documentation does not contain an equivalent to the docs here: https://www.elastic.co/guide/en/beats/metricbeat/8.4/metricbeat-module-aws.html#aws-api-requests.

So I think that pointing to the cloudwatch pricing page might be the best we can do here?

**Comment by dedemorton** [comment]
@tommyers-elastic @ravi I think we're about done with this, but I need answers to open issues listed in the [description](https://github.com/elastic/observability-docs/pull/2106#issue-1344712103).

Also I don't think I'm getting the right metrics for EC2. Might not have set things up correctly in AWS.

**Comment by ravikesarwani** [comment]
@dedemorton Tom is out this week but I would suggest for you to have a quick sync with him on the technical issues you are seeing next week. The Logs and Infra images would be nice to have and I don't see that as a complete blocker to get these changes merged for the current version of the product.

**Comment by tommyers-elastic** [comment]
hey dede - responding to the "whats left to do":

> Add a screen capture that shows the Logs app. I didn't do that yet because I see some "failed to find message" messages. Should I ignore and include a screenshot anyhow?

i hadn't seen this before, but "failed to find message" is not actually the log message, it's the Logs app telling you there is no 'message' field in the log event - weird! if i were you i'd just try to get a screen grab that didn't have these events in because it's pretty confusing. alternatively you could just show the logs in discover, which doesn't have this quirk (see below).

Screenshot 2022-09-13 at 09 49 42

> Add a screen capture of the Infrastructure app

I haven't used this screen in Kibana much either, and it also seems to have some quirks. For example in the 'hosts' view for me i see the EC2 instance with its associated metrics, but if i view "AWS -> EC2 Instances" then i see nothing. With regards to seeing the metrics in discover, do you have a data view configured that includes the data stream? In the screen grab below i'm looking at the "metrics-*" data view and then filtering on the `aws.ec2_metrics` data stream.

Screenshot 2022-09-13 at 09 53 31

**Comment by dedemorton** [comment]
@tommyers-elastic 

>if i were you i'd just try to get a screen grab that didn't have these events in because it's pretty confusing. 

Yes, I can do that for the tutorial. I think users might still be confused when they look at the view before filtering it, though.

>With regards to seeing the metrics in discover, do you have a data view configured that includes the data stream?

Yes, I had `metrics-*` selected. OK the weird thing now is that after filtering on the `aws.ec2_metrics`, I can now see all the metrics in Discover. But I think they were not showing up when I was viewing all metrics before (can't vouch for that, but it seemed that way).

**Comment by dedemorton** [comment]
Going to merge and backport this to 8.4. If there's a strong desire to get this into 8.3, I can backport it. Just let me know.

Other pending items are:

- [ ] Update this tutorial when the infrastructure app successfully shows EC2 info.
- [ ] Update the EC2 overview dashboard when we can get a screen capture that contains more reliable data.

**Pull request merged**